src/middle/Stan_math_signatures.ml

```diff
@@ -251,13 +263,16 @@ let stan_math_returntype name args =
         UnsizedType.check_compatible_arguments_mod_conv name (snd x) args )
       namematches
   in
-  if List.length filteredmatches = 0 then None
-    (* Return the least return type in case there are multiple options (due to implicit UInt-UReal conversion), where UInt<UReal *)
-  else
-    Some
-      (List.hd_exn
-         (List.sort ~compare:UnsizedType.compare_returntype
-            (List.map ~f:fst filteredmatches)))
+  match name with
+  | "reduce_sum" | "reduce_sum_static" -> Some (UnsizedType.ReturnType UReal)
```

Comment:
Maybe we can pull these string constants out, since they're littered throughout the code? I think repeating string literals throughout the codebase is somewhat error-prone.

Reply:
Good call, will do that. This will go away once we figure out the type variable thing.